### PR TITLE
PeerKit Additions

### PR DESCRIPTION
**Pull request by alexanderjarvis**
Hello!

Thanks for writing PeerKit. I made some local additions, mostly delegate methods that weren't implemented, including sending and receiving resources.

I also added a way to customise the display name as it was just the device UUID by default with no way of changing it.

Also, a way to stop transceiving that disconnects the sessions and the ability to set a new Transceiver e.g. when the display name is changed by a user.

Thanks

### CODE REVIEW

File: PeerKit/PeerKit.swift [hunk 1]
```diff
@@ -11,41 +11,52 @@ import MultipeerConnectivity
 
 // MARK: Type Aliases
 
-public typealias PeerBlock = ((peerID: MCPeerID) -> Void)
+public typealias PeerBlock = ((myPeerID:MCPeerID, peerID: MCPeerID) -> Void)
```
Comment (jpsim) [inline PeerKit/PeerKit.swift]: Could you add a space here to match the rest of the code? `myPeerID: MCPeerID`

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+!default.mode2v3
+*.perspectivev3
+!default.perspectivev3
+xcuserdata
+*.xccheckout
+*.moved-aside
+DerivedData
+*.hmap
+*.ipa
+*.xcuserstate
```
Comment (jpsim) [inline .gitignore]: Could you add a newline at the end of this file?

**Comment by jpsim** [comment]
This is fantastic, @alexanderjarvis! Thanks for the PR. I'll merge it once you address my two very minor comments.

**Comment by alexanderjarvis** [comment]
@jpsim done!

**Pull request merged**